Change statement joining in mapping and preprocess.

IanSudbery opened the pull request:
Removed ; from mapping.py and preprocess.py and added &&s where necce…ssary

IanSudbery committed:
Removed ; from mapping.py and preprocess.py and added &&s where neccessary

IanSudbery commented:
I feel like I should be testing this, but I don't know how to use the testing framework for the pipelines, if anyone can enlighten me...

Acribbs commented:
The pipelines used to be tested locally because of their size, this isnt happening now. The testing framework for this is so old that it would require significant refactoring. You could write a test to just check for the presence of semi colons, but looking at the failed github action tests, the runners are incredibly outdated and would require some work to use latest environments

IanSudbery commented:
I can obviously easily run a few things (say STAR/BWA/Fastqc/cutadapt/trimmomatic), but I'm not sure I feel up to running an exhaustive list of every mapper and every read processor, with every option, on every data type to make sure it is all consistent.

IanSudbery committed:
Missing && in appending many processing commands

IanSudbery committed:
Update cgatflow_python.yml

Workflows requires action cache to be v4. I don't really know what this means, but I updated to v4, and lets see.

Acribbs commented:
From the failures here, it looks like setup-miniconda is managing the env and the mambe env update creates issues. Might be worth removing instance of mamba env update and relying on setup-miniconda

Acribbs commented:
something like this (chatgpt so double check):
```
- name: Set up Miniconda with Mamba
  uses: conda-incubator/setup-miniconda@v2
  with:
    miniconda-version: "latest"
    python-version: ${{ matrix.python-version }}
    channels: conda-forge, bioconda, defaults
    channel-priority: true
    mamba-version: "*"
    use-mamba: true
    activate-environment: cgat-flow
    environment-file: conda/environments/cgat-flow-pipelines.yml
    auto-update-conda: true
    use-only-tar-bz2: true

```

IanSudbery committed:
check for empty mapper steps

IanSudbery committed:
Merge remote-tracking branch 'origin/{IS}_change_command_join' into {IS}_change_command_join

Acribbs committed:
updated github actions testing

Acribbs committed:
updated ci only conda env and only running linux not osx

Acribbs committed:
added pep8 to the env

Acribbs committed:
modified check_style

Acribbs committed:
removed all yields

Acribbs committed:
add rpy2, biopython, httplib2, intermine and mygene to CI environment

Acribbs committed:
add cgat-apps, toposort, beautifulsoup4, cython, and drmaa dependencies to CI environment

Acribbs committed:
accidentally copies stuff over when testing import and excluded some things

Acribbs committed:
added bashlex to ci

Acribbs committed:
ignored logfiles2tsv

Acribbs committed:
remove .py extension from cgat_logfiles2tsv in test exclusion list

Acribbs committed:
exclude conda directory from import tests

Acribbs committed:
add farm to excluded files in import tests

Acribbs committed:
remove .py extension from excluded module names in test imports

Acribbs committed:
depricated warnings for rpy2 (because peakcalling is too onld)

Acribbs committed:
removed import testing

Acribbs committed:
test for .11 and .10 only

Acribbs commented:
@IanSudbery This now passes, but the whole repo is a bit of a mess, I have updated the github actions so the runners at least work, but a lot of import tests were failing because the latest python version have a lot of deprecations. This would likely take a lot of effort to get all pipelines fully up to scratch. You happy to merge?

IanSudbery commented:
Yes. Are we going to make a descision that the pipelines are no longer maintained? Or hive off something easier to maintain that will be maintained, and bits that won't. Its something that should probably be done at some point. 

We still use the readqc and mapping pipelines, occasionally the bamstats pipeline but pretty muhc nothing else.

Acribbs commented:
I would suggest we significantly reduce the pipelines. Down to its absolute minimum

IanSudbery commented:
I agree. For me the framework is more important than the final pipelines. But I guess this is a decision we can't make just the two of us.

Acribbs commented:
Is anyone else using these pipelines though?

Acribbs commented:
likely a seperate issue, I will merge for the time being and we can come back to this
